src/WinRT.Runtime/ComWrappersSupport.cs

```diff
@@ -395,9 +395,10 @@ internal static string GetRuntimeClassForTypeCreation(IInspectable inspectable,
             return runtimeClassName;
         }
 
-        private static bool ShouldProvideIReference(Type type)
+        private readonly static ConcurrentDictionary<Type, bool> IsWindowsRuntimeTypeCache = new ConcurrentDictionary<Type, bool>();
```

Comment:
Should the caching be done at Projections.IsTypeWindowsRuntimeType instead?

Reply:
or in addition to, since they're both expensive and independent.  will do that.